src/hardware/serialport/softmodem.cpp

```diff
@@ -96,6 +96,15 @@ bool MODEM_ReadPhonebook(const std::string &filename) {
 	return true;
 }
 
+void MODEM_ClearPhonebook()
+{
+	for (auto entry : phones) {
+		delete entry;
+	}
```

Comment:
*Two suggestions, both are completely optional and are not blocking this PR from being merged.*

(Optional improvement)

This loop would be unnecessary if type of `phones` was changed from `std::vector<PhonebookEntry *>` to `std::vector<PhonebookEntry>`. This way we could also do a tiny speedup in softmodem.cpp:92 replacing

```C++
PhoneBookEntry *pbEntry = new PhonebookEntry(phone, address);
phones.push_back(pbEntry);
```
with
```
phones.emplace_back(phone, address);
```

(General comment about using auto in for-each loop in C++)

Good rule of thumb to follow is:

```C++
// xs is a container of raw pointers
for (auto *x : xs) {} // good - gives reader a hint, that xs is full of pointers
for (auto &x : xs) {} // bad - hopefully compiler can optimize it out
for (auto x : xs) {}  // works, but bad - might result in silent performance regression if container changes type during code refactor

// ys is a container of objects
for (auto *y : ys) {} // won't compile
for (auto &y : ys) {} // good
for (auto y : ys) {}  // works, but bad - will call copy constructor passing every object - use it only when it's exactly what you want

// zs is a container of simple types (e.g. ints)
for (auto *z : zs) {} // won't compile
for (auto &z : zs) {} // ok - if you want to modify the container
for (auto z : zs) {}  // ok - preferable, if you don't need to modify container
```
In other words, when dealing with container with pointers, prefer `for (auto *x : xs) …` or (not in this case) `for (const auto *x : xs) …`.

Reply:
I've originally used pointers because push_back would call a copy constructor which is slow but emplace_back seems like it solves this.